### PR TITLE
update changes in 4 files for nc-news-5

### PR DESCRIPTION
**Pull request by AndyBeeee**

### CODE REVIEW

File: __tests__/test.test.js [hunk 1]
```diff
@@ -113,5 +113,48 @@ describe('GET /api/articles/:article_id', () => {
...
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        })
+
+    test('Request responds with an article that has the correct properties and comment count', () => {
+        return request(app)
+        .get('/api/articles')
+        .then(({ body }) => {
+            expect(body.articles[6]).toMatchObject({
```
Comment (EddTheDuck) [inline __tests__/test.test.js]: You should check each article matches rather just one. Consider using a forEach.

File: __tests__/test.test.js [hunk 2]
```diff
@@ -113,5 +113,48 @@ describe('GET /api/articles/:article_id', () => {
...
+            let sorted = true
+            let date = articles[0].created_at
+            articles.forEach(article => {
+                const nextDate = article.created_at
+                if(nextDate > date) {
+                isSorted = false
+                }
+                date = nextDate
+                })
+            expect(sorted).toBe(true)
```
Comment (EddTheDuck) [inline __tests__/test.test.js]: Have a look at jest-sorted which will make testing for the order of things a lot easier.

https://www.npmjs.com/package/jest-sorted

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
If you could please make your tests a bit more robust and then re-submit.

Thanks!

File: __tests__/test.test.js [hunk 3]
```diff
@@ -113,5 +114,44 @@ describe('GET /api/articles/:article_id', () => {
...
+                article_id: expect.any(Number),
+                topic: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+                comment_count: expect.any(Number)
+              })
+            })
+        })
+    })
```
Comment (RichardNeat) [inline __tests__/test.test.js:145]: You could - if you wanted to - include a test to ensure you aren't getting the response body back. Totally optional if you wanted to be explicit.

File: db/models/articles.model.js [hunk 4]
```diff
@@ -10,4 +10,28 @@ exports.selectArticleById = (article_id) => {
             } 
             return rows[0]
     })
+}
+
```
Comment (RichardNeat) [inline db/models/articles.model.js:14]: Cool solution here Andy - we could have achieved this using SQL queries and aggregate functions to get the comment_count column. Have a go at refactoring to that if you want to. Would look a bit nicer :)

**Review by RichardNeat — APPROVED** [review]
Nice one Andy! A couple of suggested changes down below :)

**Pull request merged**